instrumentation/servlet/servlet-common/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/servlet/BaseServletHelper.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.servlet;
+
+import static io.opentelemetry.instrumentation.api.servlet.ServerSpanNaming.Source.FILTER;
+import static io.opentelemetry.instrumentation.api.servlet.ServerSpanNaming.Source.SERVLET;
+
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.SpanContext;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.instrumentation.api.instrumenter.Instrumenter;
+import io.opentelemetry.instrumentation.api.servlet.AppServerBridge;
+import io.opentelemetry.instrumentation.api.servlet.MappingResolver;
+import io.opentelemetry.instrumentation.api.servlet.ServerSpanNaming;
+import io.opentelemetry.instrumentation.api.servlet.ServletContextPath;
+import io.opentelemetry.instrumentation.api.tracer.HttpServerTracer;
+import io.opentelemetry.instrumentation.servlet.ServletAccessor;
+import io.opentelemetry.instrumentation.servlet.naming.ServletSpanNameProvider;
+
+public abstract class BaseServletHelper<REQUEST, RESPONSE> {
+  protected final Instrumenter<ServletRequestContext<REQUEST>, ServletResponseContext<RESPONSE>>
+      instrumenter;
+  protected final ServletAccessor<REQUEST, RESPONSE> accessor;
+  private final ServletSpanNameProvider<REQUEST> spanNameProvider;
+
+  protected BaseServletHelper(
+      Instrumenter<ServletRequestContext<REQUEST>, ServletResponseContext<RESPONSE>> instrumenter,
+      ServletAccessor<REQUEST, RESPONSE> accessor) {
+    this.instrumenter = instrumenter;
+    this.accessor = accessor;
+    this.spanNameProvider = new ServletSpanNameProvider<>(accessor);
+  }
+
+  public boolean shouldStart(Context parentContext, ServletRequestContext<REQUEST> requestContext) {
+    return instrumenter.shouldStart(parentContext, requestContext);
+  }
+
+  protected Context startSpan(
```

Comment:
Should we use the `shouldStart`/`start`/`end` naming (same as `Instrumenter`)? I noticed that some methods are named `startSpan`, some `startServletSpan` etc

Reply:
Yes, I think it is a good idea. Currently the names are as they were in tracers, I'll rename them.